src/actions/write/removeController.test.ts

```diff
@@ -0,0 +1,101 @@
+/**
+ * Copyright (C) 2022-2024 Permanent Data Solutions, Inc. All Rights Reserved.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+import { AntContractWriteResult } from 'src/types';
+
+import {
+  INVALID_INPUT_MESSAGE,
+  NON_CONTRACT_OWNER_MESSAGE,
+  baselineAntState,
+} from '../../../tests/utils/constants';
+import { removeController } from './removeController';
+
+describe('removeController', () => {
+  let state = { ...baselineAntState };
+
+  beforeEach(() => {
+    state = { ...baselineAntState };
+  });
```

Comment:
i would create a function and use it when you want the ANT state. you can then remove these lines.

e.g.

```
export functon generateAntBaseState = (params) => {
   return {
       ....baselineAntState
   }
}
```

and instead of 
```
const _state = { ...state, controllers: [target] };
```
use the function
```
const initState = { ...generateBaselineAntState, controllers: [target] };
const result = await removeContrller(initState);
...
```

you can then remove these lines
```suggestion
```

Reply:
the intent is to always guarantee a fresh state object that wasn't manipulated in other tests, so a function or spreading the base state would achieve that